### ISSUE TITLE
find way to avoid warnings() in 2018

### ISSUE THREAD

**Issue by taylori** [cgrandin/hake-assessment#254]
build.doc() is working great, but it would be nice if we never had warnings at the end unless something actually went wrong. Some might be inevitable, but there are a few R warnings, like the one about "zero-length arrow", which can be avoided by using commands like the following (copied from https://github.com/cgrandin/hake-assessment/blob/master/doc/r/r-functions/figures-age-comp-forecast.r#L155):

```
  # deal with warnings and make arrows
  old_warn <- options()$warn      # previous setting for warnings
  options(warn=-1)                # turn off "zero-length arrow" warning
  # add 95% intervals
  arrows(x0=0:20, x1=0:20,
         y0=apply(natsel.prop, 2, quantile, probs=0.025),
         y1=apply(natsel.prop, 2, quantile, probs=0.975),
         code=3, angle=90, length=0.05)
  # add 50% intervals
  arrows(x0=0:20, x1=0:20,
         y0=apply(natsel.prop, 2, quantile, probs=0.25),
         y1=apply(natsel.prop, 2, quantile, probs=0.75),
         code=0, angle=90, length=0, lwd=3)
  options(warn=old_warn)  #restore warnings to previous setting

```

I don't want to mess with any of the code now, so am posting this issue so that we can strive to get these things sorted out for next year.

**Comment by andrew-edwards** [comment]
The latex part runs super-clean though!

**Comment by cgrandin** [comment]
Yes, I wanted to fix that 'properly' but document has always taken priority. I really don't like the warnings either!

**Comment by taylori** [comment]
While adding values of Dirichlet-Multinomial data weighting parameters at the bottom of custom-knitr-variables.r, I noticed that the files was generating some warnings about "NAs introduced by coercion" caused by calculations related to column names being applied to non-numeric values like "unfished". 

I was able to make them go away in commit 9b569c697391c257e27f61efb841b72b64e59a62 by changing things like
```
as.numeric(names(base.model$mcmccalcs$rmed)) > 2010
```
to
```
names(base.model$mcmccalcs$rmed) %in% 2010:end.yr
```
It might have been better to deal directly with base.model$mcmccalcs$rmed and similar data.frames, but that seemed riskier.

This dropped the total from 31 warnings in `build.doc()` to just 2:
```
Warning messages:
1: package 'PBSmodelling' was built under R version 3.4.2 
2: In in_dir(input_dir(), evaluate(code, envir = env, new_device = FALSE,  :
  You changed the working directory to c:/github/hake-assessment/doc/r (probably via setwd()). It will be restored to c:/github/hake-assessment/doc. See the Note section in ?knitr::knit
```

Additional warnings occur when I clear out knitr-cache and it has to remake the figures, but those are all the same as before. We can close this issue at any time, I just wanted to get the word out about the change.
